src/Marten/Util/CollectionToArrayJsonConverter.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Marten.Util
+{
+    /// <summary>
+    /// Serialize collection type property to JSON array using a custom Newtonsoft.Json JsonConverter
+    /// Note that without using custom `JsonConverter`, `Newtonsoft.Json` stores it as $type and $value.
+    /// Or you may need to resort to `Newtonsoft.Json.TypeNameHandling.None` which has its own side-effects
+    /// </summary>
+    /// <typeparam name="T">Type of the collection</typeparam>
+    public class CollectionToArrayJsonConverter : JsonConverter
```

Comment:
I think it would make sense to rename this class to `JsonNetCollectionToArrayJsonConverter` and move it under `Marten.Services.Json` namespace